gulpfile.js

```diff
@@ -948,7 +949,10 @@ gulp.task('jsdoc', function (done) {
 
   var JSDOC_FILES = [
     'src/doc_helper.js',
+    'src/interfaces.js',
+    'src/core/primitives.js',
```

Comment:
This file makes absolutely no sense to include in API documentation, since it contains functionality which *only* runs in a web worker and isn't even accessible for an API consumer anyway (the same goes for most/all code in the `src/core/` folder); please see https://github.com/mozilla/pdf.js/pull/9369#issuecomment-358589858 for additional information.

---

Also, what's the point of *this* PR when there's already the #9369 in progress, which seems much more complete overall?

Reply:
Ok I see. Would it then make more sense to declare the interfaces that are used in the API with JSDoc so that the documentation can refer to them without risking confusing them for the datatypes used internally in the worker? 

I suppose I'd try to mock them up in `doc_helper` so that they may be corrected later.

---

Generating `.d.ts` files from JSDoc means less maintance because you don't have to remember to edit the external `.d.ts` file. Another bonus is that maintaining the JSDoc comments for dts typings would improve/structure the API documentation and vice versa.  I'm not guaranteeing that it'd be perfect, but to me it seems like it would be useful.